src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -302,6 +302,12 @@ public StorageClientFactory StorageClientFactory
             }
         }
 
+        // Test hook to set storage account. 
+        internal void SetAccountProvider(IStorageAccountProvider provider)
```

Comment:
That sounds better




On Mon, May 23, 2016 at 1:50 PM -0700, "Mike Stall" <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs<https://github.com/Azure/azure-webjobs-sdk/pull/730#discussion_r64286416>:

> @@ -302,6 +302,12 @@ public StorageClientFactory StorageClientFactory
>              }
>          }
>
> +        // Test hook to set storage account.
> +        internal void SetAccountProvider(IStorageAccountProvider provider)


How do I set the _contextFactory? The setter is private. Existing tests end up creating a whole new IServiceProvider (and skip JobHostConfiguration altogether) and go down a different path than prod.
Alternatively - I could make the ContextFactory property internal?

-
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/Azure/azure-webjobs-sdk/pull/730/files/80a4a0771ae194500bcf4c90bf74cdd7c2d5ff66#r64286416>